services/scan/src/precinct_scanner_app.ts

```diff
@@ -0,0 +1,466 @@
+import { OkResponse, Scan } from '@votingworks/api';
+import * as streams from 'memory-streams';
+import { Buffer } from 'buffer';
+import * as fsExtra from 'fs-extra';
+import {
+  BallotPageLayoutSchema,
+  BallotPageLayoutWithImage,
+  safeParse,
+  safeParseElectionDefinition,
+  safeParseJson,
+} from '@votingworks/types';
+import { assert, find } from '@votingworks/utils';
+import express, { Application } from 'express';
+import { readFile } from 'fs-extra';
+import multer from 'multer';
+import { z } from 'zod';
+import { PrecinctScannerStateMachine } from './precinct_scanner_state_machine';
+import { pdfToImages } from './util/pdf_to_images';
+import { Workspace } from './util/workspace';
+import { createInterpreter, loadLayouts } from './simple_interpreter';
+import { backup } from './backup';
+
+function sum(nums: number[]) {
+  return nums.reduce((a, b) => a + b, 0);
+}
+
+type NoParams = never;
+
+async function configureMachine(
+  machine: PrecinctScannerStateMachine,
+  workspace: Workspace
+) {
+  const { store } = workspace;
+  const electionDefinition = store.getElectionDefinition();
+  assert(electionDefinition);
+  const layouts = await loadLayouts(store);
+  assert(layouts);
+  const interpreter = createInterpreter({
+    electionDefinition,
+    ballotImagesPath: workspace.ballotImagesPath,
+    testMode: store.getTestMode(),
+    layouts,
+  });
+
+  machine.configure(store, interpreter);
+}
+
+export function buildPrecinctScannerApp(
+  machine: PrecinctScannerStateMachine,
+  workspace: Workspace
+): Application {
+  const { store } = workspace;
+  const app: Application = express();
+  const upload = multer({ storage: multer.diskStorage({}) });
+
+  app.use(express.raw());
+  app.use(express.json({ limit: '5mb', type: 'application/json' }));
+  app.use(express.urlencoded({ extended: false }));
+
+  app.get<NoParams, Scan.GetElectionConfigResponse>(
+    '/config/election',
+    (request, response) => {
+      const electionDefinition = store.getElectionDefinition();
+
+      if (request.accepts('application/octet-stream')) {
+        if (electionDefinition) {
+          response
+            .header('content-type', 'application/octet-stream')
+            .send(electionDefinition.electionData);
+        } else {
+          response.status(404).end();
+        }
+      } else {
+        response.json(electionDefinition ?? null);
+      }
+    }
+  );
+
+  app.patch<
+    NoParams,
+    Scan.PatchElectionConfigResponse,
+    Scan.PatchElectionConfigRequest
+  >('/config/election', (request, response) => {
+    const { body } = request;
+
+    if (!Buffer.isBuffer(body)) {
+      response.status(400).json({
+        status: 'error',
+        errors: [
+          {
+            type: 'invalid-value',
+            message: `expected content type to be application/octet-stream, got ${request.header(
+              'content-type'
+            )}`,
+          },
+        ],
+      });
+      return;
+    }
```

Comment:
are the config calls shared between central and precinct? Hopefully they can be?

Reply:
They aren't shared currently. The major obstacle to sharing them is that the central scanner API uses `Importer`, whereas this uses `store` directly. It will take some refactoring/cleverness to make them reusable.

Given that most of the actual logic of these endpoints is encapsulated in the store, I'm not sure there's _that_ much value to figuring out how to reuse them if it takes a lot of work. Something to investigate